fvm/blueprints/system.go

```diff
@@ -21,9 +22,17 @@ transaction {
 `
 
 // SystemChunkTransaction creates and returns the transaction corresponding to the system chunk
-// at the specified service address.
-func SystemChunkTransaction(serviceAddress flow.Address) *flow.TransactionBody {
-	return flow.NewTransactionBody().
-		SetScript([]byte(fmt.Sprintf(systemChunkTransactionTemplate, serviceAddress))).
-		AddAuthorizer(serviceAddress)
+// for the given chain.
+func SystemChunkTransaction(chain flow.Chain) (*flow.TransactionBody, error) {
+
+	contracts, err := systemcontracts.SystemContractsForChain(chain.ChainID())
+	if err != nil {
+		return nil, fmt.Errorf("could not get system contracts for chain: %w", err)
+	}
+
+	tx := flow.NewTransactionBody().
+		SetScript([]byte(fmt.Sprintf(systemChunkTransactionTemplate, contracts.Epoch.Address))).
+		AddAuthorizer(chain.ServiceAddress())
```

Comment:
@joshuahannan Given how the `FlowEpoch` constructor works, the `Heartbeat` and `Admin` resources are stored in the deployer's account, not the service account. Is that what we want or should we change the deployment flow to store these resources in the service account?

@ramtinms Will the system chunk work if the authorizer added here is not the service account?

Reply:
It doesn't matter to me where those resources are stored, if it is easier for you to store them in the service account, we can run a transaction to move them to the service account

Reply:
I'll keep the `Heartbeat` where it is, then 👍